var/spack/repos/builtin/packages/libuv-julia/package.py

```diff
@@ -17,6 +17,8 @@ class LibuvJulia(AutotoolsPackage):
     version("1.44.1", commit="1b2d16477fe1142adea952168d828a066e03ee4c")
     version("1.42.0", commit="3a63bf71de62c64097989254e4f03212e3bf5fc8")
 
+    depends_on("automake@1.16", type="build", when="@1.44.2:")
```

Comment:
🤔 it shouldn't require automake at all, the commits contain the generated files.

Reply:
Automake no, aclocal yes

Reply:
I guess the issue is that the files are from a git checkout and the timestamps are ~ arbitrary.

Reply:
Can you change it to just touch `aclocal.m4`, `Makefile.in`, and `configure`? There should really not be a dep on automake, and especially not a fixed version of it.

Reply:
I've asked the julia folks whether they can improve their release process of the libuv fork.

Reply:
@haampie I feel like touching 4 random files is sufficiently more hacky than ensuring the appropriate version is installed for a particular version of their LibUV fork. I agree that fixing this upstream is desirable, but also probably not likely in the immediate term with Julia in the midst of their 1.9 release. I don't know Autotools well enough to offer a patch here.

Reply:
It's not that hacky, it has precedence. Can you just apply this diff instead?

```diff
diff --git a/var/spack/repos/builtin/packages/libuv-julia/package.py b/var/spack/repos/builtin/packages/libuv-julia/package.py
index c1ce5ba69a..286f78d691 100644
--- a/var/spack/repos/builtin/packages/libuv-julia/package.py
+++ b/var/spack/repos/builtin/packages/libuv-julia/package.py
@@ -18,6 +18,13 @@ class LibuvJulia(AutotoolsPackage):
     version("1.44.1", commit="1b2d16477fe1142adea952168d828a066e03ee4c")
     version("1.42.0", commit="3a63bf71de62c64097989254e4f03212e3bf5fc8")
 
+    def autoreconf(self, spec, prefix):
+        # Configure files are generated, but git does not restore mtime
+        # by design. Therefore, touch files to avoid reconfigure.
+        touch("aclocal.m4")
+        touch("Makefile.in")
+        touch("configure")
+
     @property
     def libs(self):
         return find_libraries(["libuv"], root=self.prefix, recursive=True, shared=False)
```